### PR TITLE
Feature/freq

### PR DESCRIPTION
**Pull request by HamidehGMAO**
## Description

In OASIM we were not reading the frequency bands from the observations data, and oasim just runs with the oasim's "33" fixed bands https://github.com/JCSDA-internal/oasim/blob/c4b6659ceba65ba40ca457a72e9f85510a560945/src/oasim/oasim_constants_mod.f90#L20
here we  read the frequency for each channel and calculate the corresponding hofx

## Issue(s) addressed

[Resolves #<issue_number>](https://github.com/JCSDA-internal/ufo/issues/2803)

## Dependencies

there is a corresponding PR in UFO, they need to merge at the sam time

### CODE REVIEW

File: src/oasim/rlwn_mod.f90 [hunk 1]
```diff
@@ -81,8 +82,11 @@ subroutine rlwn(km, lam, cosz, aw, bw, ac, bc, bpic, WtoQ, Ed, Es, h, phyto, exc
 call edeu(km, lam, aw, bw, ac, bc, bpic, WtoQ, Ed, Es, H, p, excdom, exdet, rmud, tirrq, cdomabsq, &
           avgq, sfceu)
 
-rn2 = rn*rn      !index refr squared  
-rlwnref = (1.0-rho)*sfceu/(rn2*Q)
+do i = 1,size(l_chan)
+   ind = MINLOC(abs(lam-l_chan(i)), DIM=1)
+   rn2 = rn*rn      !index refr squared
```
Comment (kbhargava) [inline src/oasim/rlwn_mod.f90]: Does this need to move inside the loop? NOt sure if the `rn` values depend on `i` here

**Review by travissluka — APPROVED** [review]
please address @kbhargava 's comment, otherwise looks fine and seems to be working as intended.

File: src/oasim/rlwn_mod.f90 (hunk 1 above)
Comment (HamidehGMAO) [reply src/oasim/rlwn_mod.f90]: you are right @kbhargava ! I just fixed it! Thanks

**Comment by HamidehGMAO** [comment]
Thanks @travissluka and @kbhargava for your reviews, I just addressed Kriti's comment.

**Review by kbhargava — APPROVED** [review]
👍🏽

**Pull request merged**